### PR TITLE
`mixed` `ReflectionNamedType` does allow `null` values

### PR DESCRIPTION
**Pull request by ondrejmirtes**
Matches PHP core behaviour: https://3v4l.org/FO5e4

### CODE REVIEW

**Comment by Ocramius** [comment]
@ondrejmirtes could you rebase this onto `5.0.x`? I'd tag a patch release afterwards :+1:

**Comment by ondrejmirtes** [comment]
Rebased :) And at least one more fix is coming, once I figure out something :)

File: src/Reflection/Adapter/ReflectionNamedType.php [hunk 1]
```diff
@@ -23,7 +25,7 @@ public function getName(): string
 
     public function __toString(): string
     {
-        return ($this->allowsNull ? '?' : '')
+        return ($this->allowsNull && strtolower($this->betterReflectionType->getName()) !== 'mixed' ? '?' : '')
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionNamedType.php:28]: This is kinda missing a test, but meh :P

**Review by Ocramius — APPROVED** [review]
Thanks @ondrejmirtes!

File: src/Reflection/Adapter/ReflectionNamedType.php (hunk 1 above)
Comment (ondrejmirtes) [reply src/Reflection/Adapter/ReflectionNamedType.php:28]: Some tests failed with not expecting `?mixed` without this change :)

Comment (Ocramius) [reply src/Reflection/Adapter/ReflectionNamedType.php:28]: All good then :D :muscle:

**Pull request merged**